### PR TITLE
fixed up the settings

### LINKED ISSUE #21: Clean up the start_menu function
The start_menu function contains inside the logic for the settings. We need to separate the latter from the former, and pull out all the settings-related code into its own function that gets called in start_menu.
### ACCEPTANCE CRITERIA:

- Create a new `main_settings` function which contains all the code related to game settings from the `start_menu`
- Add an entry to this new function to change the text speed
- Have each setting be its own proper function, so that `main_settings` does not get cluttered with the logic of all the different settings
- The start_menu function does not contain any settings-related code apart from a call to the `main_settings` function

### PR DESCRIPTION
**Pull request by DIONYSUS-BASOUNAS**
closes #21 
if you have any objection or any additions don't hesitate to correct me

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -5,5 +5,8 @@
 		"editor.semanticHighlighting.enabled": true,
 		"editor.formatOnSave": true,
 		"editor.formatOnPaste": true
+	},
+	"files.associations": {
+		"iostream": "cpp"
```
Comment (DIONYSUS-BASOUNAS) [inline .vscode/settings.json:10]: i fixed it

File: src/start_menu/main_settings/weight.h [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#pragma once
+#include "../start_menu.h"
```
Comment (achilleasmitos) [inline src/start_menu/main_settings/weight.h:2]: This include should not be needed here.

File: src/start_menu/main_settings/speed.h [hunk 3]
```diff
@@ -0,0 +1,4 @@
+#pragma once
+#include "../start_menu.h"
+
+void Speed();
```
Comment (achilleasmitos) [inline src/start_menu/main_settings/speed.h:4]: Provide an empty line at the end of the .cpp and .h files.

Also, more descriptive names and comments above the declarations (see weight.h for similar comment).

File: src/start_menu/main_settings/speed.cpp [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#include "speed.h"
+#include "../../utils/utils.h"
+#include <iostream>
+
+void Speed(){
+	std::cout << "Give me the time you want between each rendered "
+				 "character (in ms): ";
```
Comment (achilleasmitos) [inline src/start_menu/main_settings/speed.cpp:7]: First write the commands in single line, then format them.

File: src/start_menu/main_settings/speed.cpp [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+	utils::GetUserInput(utils::g_sleep_for_ms);
+}
```
Comment (achilleasmitos) [inline src/start_menu/main_settings/speed.cpp:9]: Empty line at the end of the file.

File: src/start_menu/main_settings/size.h
Comment (achilleasmitos) [inline src/start_menu/main_settings/size.h:1]: Follow the comments on width.h for this one too.

File: src/start_menu/main_settings/size.cpp [hunk 6]
```diff
@@ -0,0 +1,9 @@
+#include "size.h"
+#include "../../utils/utils.h"
+#include <iostream>
+
+void Size(){
+	std::cout << "Pick Number" << std::endl;
```
Comment (achilleasmitos) [inline src/start_menu/main_settings/size.cpp:6]: Perhaps some more words here to inform the user of what that 'number' represents?

File: src/start_menu/main_settings/size.cpp [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
+	utils::GetUserInput(utils::g_text_size);
+	utils::CustomizeText(utils::g_text_size);
+}
```
Comment (achilleasmitos) [inline src/start_menu/main_settings/size.cpp:9]: Empty line at the EOF.

File: src/start_menu/main_settings/main_settings.h
Comment (achilleasmitos) [inline src/start_menu/main_settings/main_settings.h:1]: Follow the comments on width.h for this one too.

File: src/start_menu/main_settings/main_settings.cpp [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+#include <iostream>
+
+void MainSettings()
+{
+	int go_back = 0;
+	do
+	{
+		// code for settings
+		std::cout << "Opening settings..." << std::endl;
+		std::cout << "chouse a setting you want to change" << std::endl;
```
Comment (achilleasmitos) [inline src/start_menu/main_settings/main_settings.cpp:16]:
```suggestion
		std::cout << "Choose the setting that you want to change" << std::endl;
```

File: src/start_menu/main_settings/main_settings.cpp [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+				{
+					std::cout << "Invalid choice! Please "
+								 "choose again."
+							  << std::endl;
+					go_back = 0;
+					continue;
+				}
+		}
+	}
+	while (go_back == 0);
```
Comment (achilleasmitos) [inline src/start_menu/main_settings/main_settings.cpp:60]: I believe we can get rid of the `go_back` var. You have already done so in other files with `while(true)`

File: src/start_menu/main_settings/main_settings.cpp [hunk 10]
```diff
@@ -0,0 +1,61 @@
+#include "weight.h"
+#include "speed.h"
+#include "size.h"
+#include "main_settings.h"
```
Comment (achilleasmitos) [inline src/start_menu/main_settings/main_settings.cpp:4]: The `include` of this files header should always be on top. The rest of them should follow alphabetically.

File: src/start_menu/main_settings/font.cpp [hunk 11]
```diff
@@ -0,0 +1,5 @@
+#include "font.h"
+#include "../../utils/utils.h"
+#include <iostream>
+
+void Font() { std::cout << "no idea what to do..." << std::endl; }
```
Comment (achilleasmitos) [inline src/start_menu/main_settings/font.cpp:5]: EOF empty line.

**Review by achilleasmitos — CHANGES_REQUESTED** [review]
We need more consistency with how we handle infinite iterations (`do...while`), comments in header files, includes, and empty lines at the end of files. 
Also, please remember to format your files before committing them (it should be automatic for you upon save, which is a setting in settings.json; make sure that you don't overwrite this in your local VSCode).

File: .vscode/settings.json [hunk 12]
```diff
@@ -5,5 +5,6 @@
 		"editor.semanticHighlighting.enabled": true,
 		"editor.formatOnSave": true,
 		"editor.formatOnPaste": true
-	}
+	},
+
```
[the remark this answers is not in the document]
Comment (DIONYSUS-BASOUNAS) [reply .vscode/settings.json:9]: done

File: src/start_menu/main_settings/weight.cpp [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+{
+	do
+	{
+		std::cout << "Pick Weight" << std::endl;
+		std::cout << "1. light" << std::endl;
+		std::cout << "2. SemiLight" << std::endl;
+		std::cout << "3. Normal" << std::endl;
+		std::cout << "4. SemiBold" << std::endl;
+		std::cout << "5. Bold" << std::endl;
+		int weight_shoice;
```
[the remark this answers is not in the document]
Comment (DIONYSUS-BASOUNAS) [reply src/start_menu/main_settings/weight.cpp:15]: done

File: src/start_menu/main_settings/weight.h (hunk 2 above)
Comment (DIONYSUS-BASOUNAS) [reply src/start_menu/main_settings/weight.h:2]: done

File: src/start_menu/main_settings/speed.h [hunk 14]
```diff
@@ -0,0 +1,4 @@
+#pragma once
+#include "../start_menu.h"
```
[the remark this answers is not in the document]
Comment (DIONYSUS-BASOUNAS) [reply src/start_menu/main_settings/speed.h:2]: done

File: src/start_menu/main_settings/speed.cpp (hunk 4 above)
Comment (DIONYSUS-BASOUNAS) [reply src/start_menu/main_settings/speed.cpp:7]: done

File: src/start_menu/main_settings/speed.cpp (hunk 5 above)
Comment (DIONYSUS-BASOUNAS) [reply src/start_menu/main_settings/speed.cpp:9]: done

File: src/start_menu/main_settings/size.cpp (hunk 7 above)
Comment (DIONYSUS-BASOUNAS) [reply src/start_menu/main_settings/size.cpp:9]: done

File: src/start_menu/main_settings/main_settings.cpp [hunk 15]
```diff
@@ -0,0 +1,61 @@
...
+					std::cout << "Invalid choice! Please "
+								 "choose again."
+							  << std::endl;
+					go_back = 0;
+					continue;
+				}
+		}
+	}
+	while (go_back == 0);
+}
```
[the remark this answers is not in the document]
Comment (DIONYSUS-BASOUNAS) [reply src/start_menu/main_settings/main_settings.cpp:61]: done

File: src/start_menu/main_settings/font.cpp (hunk 11 above)
Comment (DIONYSUS-BASOUNAS) [reply src/start_menu/main_settings/font.cpp:5]: done

File: src/start_menu/main_settings/main_settings.cpp (hunk 8 above)
Comment (DIONYSUS-BASOUNAS) [reply src/start_menu/main_settings/main_settings.cpp:16]: done

File: src/start_menu/main_settings/main_settings.cpp [hunk 16]
```diff
@@ -0,0 +1,61 @@
...
+	{
+		// code for settings
+		std::cout << "Opening settings..." << std::endl;
+		std::cout << "chouse a setting you want to change" << std::endl;
+		std::cout << "Please select an option:" << std::endl;
+		std::cout << "1. Text Size" << std::endl;
+		std::cout << "2. Font Weight" << std::endl;
+		std::cout << "3. Font Name" << std::endl;
+		std::cout << "4. Text Speed" << std::endl;
+		int setting_shoice;
```
[the remark this answers is not in the document]
Comment (DIONYSUS-BASOUNAS) [reply src/start_menu/main_settings/main_settings.cpp:22]: done

File: src/start_menu/main_settings/weight.h [hunk 17]
```diff
@@ -0,0 +1,5 @@
...
+
+
```
[the remark this answers is not in the document]
Comment (DIONYSUS-BASOUNAS) [reply src/start_menu/main_settings/weight.h:4]: done

File: src/start_menu/main_settings/speed.h (hunk 3 above)
Comment (DIONYSUS-BASOUNAS) [reply src/start_menu/main_settings/speed.h:4]: done

File: src/start_menu/main_settings/size.h
Comment (DIONYSUS-BASOUNAS) [reply src/start_menu/main_settings/size.h:1]: done

File: src/start_menu/main_settings/size.cpp (hunk 6 above)
Comment (DIONYSUS-BASOUNAS) [reply src/start_menu/main_settings/size.cpp:6]: done

File: src/start_menu/main_settings/main_settings.h
Comment (DIONYSUS-BASOUNAS) [reply src/start_menu/main_settings/main_settings.h:1]: done

File: src/start_menu/main_settings/font.h
[the remark this answers is not in the document]
Comment (DIONYSUS-BASOUNAS) [reply src/start_menu/main_settings/font.h:1]: done

File: src/start_menu/main_settings/weight.cpp [hunk 18]
```diff
@@ -0,0 +1,51 @@
...
+		utils::GetUserInput(weight_shoice);
+		if (weight_shoice == 1)
```
[the remark this answers is not in the document]
Comment (DIONYSUS-BASOUNAS) [reply src/start_menu/main_settings/weight.cpp:17]: done

File: src/start_menu/main_settings/main_settings.cpp (hunk 9 above)
Comment (DIONYSUS-BASOUNAS) [reply src/start_menu/main_settings/main_settings.cpp]: done

File: .vscode/settings.json [hunk 19]
```diff
@@ -5,5 +5,70 @@
 		"editor.semanticHighlighting.enabled": true,
 		"editor.formatOnSave": true,
 		"editor.formatOnPaste": true
+	},
```
Comment (achilleasmitos) [inline .vscode/settings.json:8]: Again, this file should remain unchanged

File: src/start_menu/main_settings/ChangeFontWeight.cpp [hunk 20]
```diff
@@ -0,0 +1,55 @@
...
+					utils::CustomizeText(utils::g_text_size, utils::Bold);
+					break;
+				}
+			default:
+				{
+					std::cout << "Invalid choice! Please choose again." << std::endl;
+					continue;
+				}
+		}
+		break;
```
Comment (achilleasmitos) [inline src/start_menu/main_settings/ChangeFontWeight.cpp:52]: Not needed, I believe.

File: src/start_menu/main_settings/ChangeFontWeight.h [hunk 21]
```diff
@@ -0,0 +1,3 @@
+#pragma once
+
+void Weight();//Declares the Weight() function to prompt the user for a weight choice and customize text weight
```
Comment (achilleasmitos) [inline src/start_menu/main_settings/ChangeFontWeight.h:3]: Change the function's name to match the name of the file.

Also, see how comments for functions are done in other header files (with /***/ before the declaration)

File: src/start_menu/main_settings/main_settings.cpp [hunk 22]
```diff
@@ -0,0 +1,56 @@
...
+					Speed(); // Call the Speed() function to prompt the user for the time between each rendered character
+					break;
+				}
+			default:
+				{
+					std::cout << "Invalid choice! Please choose again." << std::endl;
+					continue;
+				}
+		}
+		break;
```
Comment (achilleasmitos) [inline src/start_menu/main_settings/main_settings.cpp:53]: I believe this is unnecessary.

File: src/start_menu/start_menu.cpp [hunk 23]
```diff
@@ -1,11 +1,11 @@
 #include "start_menu.h"
+#include "./main_settings/main_settings.h"
 #include "../utils/utils.h"
 #include <iostream>
 
 int start_menu::StartMenu()
 {
-	int go_back = 0;
-	int choice = -1;
+		int choice = -1;
```
Comment (achilleasmitos) [inline src/start_menu/start_menu.cpp:8]: Has this been formatted with clang? There is an extra tab here...

**Review by achilleasmitos — CHANGES_REQUESTED** [review]
Pay more attention to detail. Review your changes (with `git diff HEAD~` after you have committed them) before pushing them.

File: src/start_menu/main_settings/main_settings.cpp (hunk 22 above)
Comment (DIONYSUS-BASOUNAS) [reply src/start_menu/main_settings/main_settings.cpp:53]: from what i have seen it is needed, if you found a way without it pls inform me

File: src/start_menu/main_settings/ChangeFontWeight.cpp (hunk 20 above)
Comment (achilleasmitos) [reply src/start_menu/main_settings/ChangeFontWeight.cpp:52]: it is necessary after all. I was confused with the breaks inside the switch statements

**Review by achilleasmitos — COMMENTED** [review]
Nice. All that's left to do now is to change the names of the other functions, together with their respective files, like we did with `ChangeFontWeight`. The functions in question are: `Font`, `Size`, and `Speed`. They should be something like `ChangeTextFont`, `ChangeTextSize` and `ChangeTextSpeed` respectively.

If you are feeling generous, you could also change the `ChangeFontWeight` to `ChangeTextWeight` to match with the naming of the rest of the functions.

**Review by achilleasmitos — APPROVED** [review]
Nice job! Go ahead and squash-merge the changes.

**Pull request merged**